traffic_ops/traffic_ops_golang/perlhash.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+*/
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+func ParsePerlObj(s string) (map[string]interface{}, error) {
+	obj, _, err := getObj(s)
+	return obj, err
+}
+
+func getObj(s string) (map[string]interface{}, string, error) {
+	obj := map[string]interface{}{}
+
+	s = strings.TrimSpace(s)
+	if len(s) < 1 || s[0] != '{' {
+		return obj, "", fmt.Errorf("expected first character '{': %v", s)
+	}
+	s = s[1:] // strip opening {
+	s = strings.TrimSpace(s)
+
+	// read top-level keys
+	for {
+		s = stripComment(s)
+		s = strings.TrimSpace(s)
+		// s = stripComment(s)
+		if len(s) > 0 && s[0] == '}' {
+			return obj, s[1:], nil
+		}
+
+		key := ""
+		key, s = getKey(s)
+
+		s = strings.TrimSpace(s)
+		if len(s) == 0 {
+			return obj, "", fmt.Errorf("malformed string after key '%v'", key)
+		}
+
+		err := error(nil)
+		switch {
+		case s[0] == '{':
+			v := map[string]interface{}{}
+			v, s, err = getObj(s)
+			if err != nil {
+				return obj, "", fmt.Errorf("Error getting object value after key %v: %v", key, err)
+			}
+			obj[key] = v
+		case s[0] == '\'':
+			v := ""
+			v, s, err = getStr(s)
+			if err != nil {
+				return obj, "", fmt.Errorf("Error getting string value after key %v: %v", key, err)
+			}
+			obj[key] = v
+		case unicode.IsDigit(rune(s[0])):
```

Comment:
Eh, I care, but it'd be very expensive time-wise to fix properly. I'm sure there are other cases (in fact, we know it doesn't support quoted keys, or double-quoted values). But it works for the Perl config as generated by Postinstall. And since it's only temporary until Perl goes away, I'm voting we don't further delay Step 0 of the migration.

If we hit issues in the future, we can always go back and make the parser more robust.